drivers/hmis/app/graphql/types/base_object.rb

```diff
@@ -49,6 +49,18 @@ def load_ar_association(object, association, scope: nil)
       dataloader.with(Sources::ActiveRecordAssociation, association, scope).load(object)
     end
 
+    def load_ar_scope(scope:, id:)
+      dataloader.with(Sources::ActiveRecordScope, scope).load(id)
+    end
+
+    def load_last_user_from_versions(object)
+      versions = load_ar_association(object, :versions)
+      last_user_id = versions.max_by(&:created_at)&.whodunnit
+      return unless last_user_id && last_user_id =~ /\A[0-9]+\z/
+
+      load_ar_scope(scope: Hmis::User.all, id: last_user_id)
```

Comment:
Nice. Loading all the versions into memory feels a bit scary, but I think the alternative is to not use the data loader, which would probably be worse.

I think elsewhere we use `load_ar_association` in a way it probably can't handle -- loading relations that have a scope condition on them. So for example if we added a "relation" `has_one :latest_version, -> { order(created_at: :desc) }` and then tried to do `load_ar_association(object, :latest_version)`. I did that elsewhere, like [here](https://github.com/greenriver/hmis-warehouse/blob/4f8c89aae6d80e403ec646ae5e291934f0a691ad/drivers/hmis/app/graphql/types/hmis_schema/merge_audit_event.rb#L31)/[here](https://github.com/greenriver/hmis-warehouse/blob/4f8c89aae6d80e403ec646ae5e291934f0a691ad/drivers/hmis/app/models/hmis/client_merge_audit.rb#L13-L14), but I have a feeling that it's not actually using the data loader effectively.  I'm curious if you have any immediate thoughts about that approach 👀

Reply:
Thanks, yeah I was thinking there might be less than 100 version records here but maybe that's not a good assumption. I narrowed the selection on the versions to avoid loading the version object/changes and moved the order to the db as you suggest, that should be more performant.

I'll review the usage of `load_ar_association()` you mentioned. It feels like you'd need to group by versions.item to make that work in a batch but I'm probably missing something